code/ndarray.c

```diff
@@ -1986,8 +1986,15 @@ mp_obj_t ndarray_info(mp_obj_t obj_in) {
 MP_DEFINE_CONST_FUN_OBJ_1(ndarray_info_obj, ndarray_info);
 #endif
 
+// (the get_buffer protocol returns 0 for success, 1 for failure)
 mp_int_t ndarray_get_buffer(mp_obj_t self_in, mp_buffer_info_t *bufinfo, mp_uint_t flags) {
     ndarray_obj_t *self = MP_OBJ_TO_PTR(self_in);
-    // buffer_p.get_buffer() returns zero for success, while mp_get_buffer returns true for success
-    return !mp_get_buffer(self->array, bufinfo, flags);
+    if (self->ndim != 1 || self->strides[0] > 1) {
+        // For now, only allow fetching buffer of a 1d-array
```

Comment:
This should be OK, though, while we can't specify the shape anyway. Once https://github.com/v923z/micropython-ulab/pull/305 is done, one could initialise from a buffer with `shape` and `strides`.